server/app/api/v3.py

```diff
@@ -323,6 +404,20 @@ def register_api_v3(app, app_config, api_url_prefix):
 
         # Custom CXGs (only for "w" dataroot)
         if url_dataroot == "w":
+            # Add authorization check before processing any /w/ requests
+            @app.before_request
```

Comment:
Safety belt: before serving `/w/...` we now demand `_resolve_request_user_id` succeeds. That way BYOD datasets remain protected even if someone deploys explorer-only mode with auth disabled.